Sources/SwiftParser/Expressions.swift

```diff
@@ -248,6 +249,8 @@ extension Parser {
   ///     type-casting-operator → 'as' type
   ///     type-casting-operator → 'as' '?' type
   ///     type-casting-operator → 'as' '!' type
+  ///     avo-pattern-matching-expression → avo-pattern-matching-operator pattern
+  ///     avo-pattern-matching-operator  →  'is' 'case'
```

Comment:
(A call for suggestions for better names, with a detour into some linguistic stuff)

With this new syntax included, we have 3 pattern matching constructs in Swift:

- `~=`
- `case`
- `is case`

They all have the same capabilities, with their only difference being the "word" order:

```swift
pattern ~= value
case pattern = value
value is case pattern
```

So it seems to me the names of the operators and expressions has to reflect this difference in order to be distinct from each other. However, the dominant order alone isn't enough to differentiate between them, because both `pattern ~= value` and `value is case pattern` have the subject-verb-object structure (if we take both operators `~=` and `is case` as a form of the English word "match").

In the comments here and that in `CodeGeneration/Sources/SyntaxSupport/ExprNodes.swift`, I'm assuming the matching is always done by the value, making it the agent. And by this logic, I have tentatively named the `is case` operator as the "agent-verb-object pattern matching operator", and the corresponding expression as "agent-verb-object pattern matching expression. This also makes `~=` "subject-verb-agent", and `case` "verb-object-agent".

However, this assumption is incorrect, because the value doesn't initiate/perform the action, so it isn't the agent.

<details>

<summary>Even longer linguistic detours (sorry)</summary>

We can first observe that all 4 sentences below are perfectly natural English sentences:

> The value matches the pattern.
> The pattern matches the value.
> The pattern is matched by the value.
> The value is matched by the pattern.

These 4 sentences make it as if both "value" and "pattern" can be the agent. However, if we use "match" as a ditransitive verb, we can see that neither the value nor the pattern is the agent.

> The computer/runtime matches the value to the pattern.

Instead, they're both themes.

</details>

Anyway, it would be great if anyone could suggest some better names for the operators and expressions.

Reply:
renamed to "new pattern-matching [operator|expression]" for now in d1afd93.

Reply:
After a discussion with Jordan, it now seems right to me to just call it "pattern-matching boolean expression", as it is in the pitch. Updated in 2bd28cc.